pkg/client/events/query_client.go

```diff
@@ -236,40 +291,30 @@ func (eqc *eventsQueryClient) goPublishEventsBz(
 			// | this method return the same error.
 
 			// Only propagate error if it's not a context cancellation error.
-			if !errors.Is(ctx.Err(), context.Canceled) {
+			if !errors.Is(evtConn.ctx.Err(), context.Canceled) {
 				// Populate the error side (left) of the either and publish it.
-				eventsBzPublishCh <- either.Error[[]byte](err)
+				evtConn.eventsBzPublishCh <- either.Error[[]byte](err)
 			}
 
-			eqc.close()
+			// cancel the context to stop the unsubscribe goroutine
+			evtConn.cancelCtx()
```

Comment:
Where does this get closed if there's no error?

Reply:
It's never closed, because this is a long living connection that needs to exist as long as the `RelayMiner` is running.